src/data/equiv/fintype.lean

```diff
@@ -0,0 +1,71 @@
+/-
+Copyright (c) 2021 Yakov Pechersky. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yakov Pechersky
+-/
+
+import data.equiv.basic
+import data.fintype.basic
+import group_theory.perm.sign
+
+/-! # Equivalence between fintypes
+
+This file contains some basic results on equivalences where one or both
+sides of the equivalence are `fintype`s.
+
+# Main definitions
+
+ - `function.embedding.to_equiv_range`: computably turn an embedding of a
+   fintype into an `equiv` of the domain to its range
+ - `equiv.perm.via_embedding : perm α → (α ↪ β) → perm β` extends the domain of
+   a permutation, fixing everything outside the range of the embedding
+
+# Implementation details
+
+ - `function.embedding.to_equiv_range` uses a computable inverse, but one that has poor
+   computational performance, since it operates by exhaustive search over the input `fintype`s.
+-/
+
+variables {α β : Type*} [fintype α] [decidable_eq β] (e : equiv.perm α) (f : α ↪ β)
+
+/--
+Computably turn an embedding `f : α ↪ β` into an equiv `α ≃ set.range f`,
+if `α` is a `fintype`. Has poor computational performance, due to exhaustive searching in
+constructed inverse. When a better inverse is known,
+use `equiv.set.range_of_left_inverse` instead.
+-/
```

Comment:
It would be good to mention that this is a computable version of `equiv.set.range`

Reply:
By the way, what's the difference between `equiv.set.range` and `equiv.of_injective`?

Reply:
Nice catch; `of_injective` was added in #3125, `equiv.set.range` was added way before that in bd013e8089378e8057dc7e93c9eaf2c8ebaf25a2.

I think we should just eliminate `equiv.set.range`, the other name is better and consistent with `alg_equiv.of_injective` etc. Mind opening a new PR for that?